goal_src/engine/nav/path.gc

```diff
@@ -5,3 +5,496 @@
+        #t
+        (bucket-id debug-draw1)
+        (the-as vector s4-1)
+        (the-as vector (&+ (-> obj curve cverts) (* (+ s5-1 1) 16)))
+        (new 'static 'rgba :r #xff :g #x80 :a #x80)
+        #f
+        (the-as rgba -1)
+        )
+       )
+      (if (nonzero? (logand (-> obj flags) (path-control-flag draw-point)))
+       (add-debug-x
+        #t
+        (bucket-id debug-draw1)
+        (the-as vector s4-1)
+        (new 'static 'rgba :r #xff :a #x80)
+        )
+       )
+      (when (nonzero? (logand (-> obj flags) (path-control-flag draw-text)))
+       (let ((s3-1 add-debug-text-3d)
+             (s2-1 #t)
+             (s1-0 68)
+             )
+        (format (clear *temp-string*) "~D" s5-1)
+        (s3-1
+         s2-1
+         (the-as bucket-id s1-0)
+         *temp-string*
+         (the-as vector s4-1)
+         (new 'static 'rgba :r #x4)
+         (the-as vector2h #f)
+         )
+        )
+       )
+      )
+     )
+    )
+   )
+  0
+  (none)
+  )
+
+;; definition for method 16 of type path-control
+(defmethod path-distance path-control ((obj path-control))
+  (let ((f30-0 0.0))
+   (dotimes (s5-0 (+ (-> obj curve num-cverts) -1))
+    (+!
+     f30-0
+     (vector-vector-distance
+      (the-as vector (&+ (-> obj curve cverts) (* s5-0 16)))
+      (the-as vector (&+ (-> obj curve cverts) (* (+ s5-0 1) 16)))
+      )
+     )
+    )
+   f30-0
+   )
+  )
+
+;; definition for method 16 of type curve-control
+(defmethod path-distance curve-control ((obj curve-control))
+  (let ((f0-0 (-> obj curve length)))
+   (when (= f0-0 0.0)
+    (set! f0-0 (curve-length (-> obj curve)))
+    (set! (-> obj curve length) f0-0)
+    )
+   f0-0
+   )
+  )
+
+;; definition for method 10 of type path-control
+;; Used lq/sq
+(defmethod
+  eval-path-curve-div!
+  path-control
+  ((obj path-control) (arg0 vector) (arg1 float) (arg2 symbol))
+  (let ((a1-1 (-> obj curve num-cverts))
+        (f0-3 (the float (the int arg1)))
+        )
+   (cond
+    ((< arg1 0.0)
+     (set!
+      (-> arg0 quad)
+      (-> (the-as (pointer uint128) (&+ (-> obj curve cverts) 0)))
+      )
+     )
+    ((>= f0-3 (the float (+ a1-1 -1)))
+     (set!
+      (-> arg0 quad)
+      (->
+       (the-as (pointer uint128) (&+ (-> obj curve cverts) (* (+ a1-1 -1) 16)))
+       )
+      )
+     )
+    (else
+     (if (or (= arg2 'exact) (= f0-3 arg1))
+      (set!
+       (-> arg0 quad)
+       (->
+        (the-as
+         (pointer uint128)
+         (&+ (-> obj curve cverts) (* (the int f0-3) 16))
+         )
+        )
+       )
+      (vector-lerp!
+       arg0
+       (the-as vector (&+ (-> obj curve cverts) (* (the int f0-3) 16)))
+       (the-as vector (&+ (-> obj curve cverts) (* (the int (+ 1.0 f0-3)) 16)))
+       (- arg1 f0-3)
+       )
+      )
+     )
+    )
+   )
+  arg0
+  )
+
+;; definition for method 11 of type path-control
+;; Used lq/sq
+(defmethod get-random-point path-control ((obj path-control) (arg0 vector))
+  (with-pp (cond
```

Comment:
The formatting of this will probably change with the PR I merged a few minutes ago.